input/cql/DiabetesEyeExamFHIR.cql

```diff
@@ -0,0 +1,111 @@
+valueset "Office Visit": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1001'
+valueset "Ophthalmological Services": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.526.3.1285'
+valueset "Preventive Care Services Established Office Visit, 18 and Up": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1025'
+valueset "Preventive Care Services Initial Office Visit, 18 and Up": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1023'
+valueset "Retinal or Dilated Eye Exam": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.115.12.1088'
+valueset "Telephone Visits": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1080'
+
+parameter "Measurement Period" Interval<DateTime>
+
+context Patient
+
+/*
+@QICore6: Update Condition to ConditionProblemsHealthConcerns
+*/
+define "Initial Population":
+  AgeInYearsAt(date from 
+    end of "Measurement Period"
+  ) in Interval[18, 75]
+    and exists ( "Qualifying Encounters" )
+    and exists ( [Condition: "Diabetes"] Diabetes
+        where Diabetes.prevalenceInterval ( ) overlaps day of "Measurement Period"
+    )
+
+define "Qualifying Encounters":
+  ( ( ["Encounter": "Office Visit"]
+      union ["Encounter": "Annual Wellness Visit"]
+      union ["Encounter": "Preventive Care Services Established Office Visit, 18 and Up"]
+      union ["Encounter": "Preventive Care Services Initial Office Visit, 18 and Up"]
+      union ["Encounter": "Home Healthcare Services"]
+      union ["Encounter": "Ophthalmological Services"]
+      union ["Encounter": "Telephone Visits"]
+  ).isEncounterPerformed ( ) ) ValidEncounters
+    where ValidEncounters.period during day of "Measurement Period"
+
+define "Denominator":
+  "Initial Population"
+
+define "Denominator Exclusions":
+  Hospice."Has Hospice Services"
+    or AIFrailLTCF."Is Age 66 or Older with Advanced Illness and Frailty"
+    or AIFrailLTCF."Is Age 66 or Older Living Long Term in a Nursing Home"
+    or PalliativeCare."Has Palliative Care in the Measurement Period"
+
+define "Numerator":
+  ( "Diabetic Retinopathy Overlapping Measurement Period"
+      and exists ( "Retinal Exam in Measurement Period" )
+  )
+    or ( not ( "Diabetic Retinopathy Overlapping Measurement Period" )
+        and exists ( "Retinal Exam in Measurement Period or Year Prior" )
+    )
+
+/*
+@QICore6: Update Observation to ObservationScreeningAssessment
+*/
+define "Retinal Exam in Measurement Period":
+  ( ( [Observation: "Retinal or Dilated Eye Exam"] ).isAssessmentPerformed ( ) ) RetinalExam
+    where RetinalExam.effective.toInterval ( ) during day of "Measurement Period"
+
+/*
+@QICore6: Update Observation to ObservationScreeningAssessment
+*/
+define "Retinal Exam in Measurement Period or Year Prior":
+  ( ( [Observation: "Retinal or Dilated Eye Exam"] ).isAssessmentPerformed ( ) ) RetinalExam
```

Comment:
Update as well based on Bryn's response.

Reply:
I updated the analysis to include **ObservationClinicalResult** (rather than ObservationScreeningAssessment) and category of **exam** (rather than survey) since the .isAssessmentPerformed function only pertains to SimpleObservation and ObservationScreeningAssessment